kcwarden/auditors/realm/abstract_realm_auditor.py

```diff
@@ -0,0 +1,20 @@
+import abc
+from typing import Generator
+
+from kcwarden.api import Auditor
+from kcwarden.custom_types.keycloak_object import Realm
+from kcwarden.custom_types.result import Result
+
+
+class AbstractRealmAuditor(Auditor, abc.ABC):
+    def should_consider_realm(self, realm: Realm) -> bool:
+        return self.is_not_ignored(realm)
+
+    def audit(self) -> Generator[Result, None, None]:
+        for realm in self._DB.get_all_realms():
+            if self.should_consider_realm(realm):
+                yield from self.audit_realm(realm)
```

Comment:
This is smart, considering that there will likely be no other reasons to exclude a realm, but it changes how realm auditors are implemented compared to all other auditors (where the method is to override a common `audit` function). Is this global inconsistency worth the local code improvement?

Reply:
I'm a friend of the DRY principle and it seems useless to have the exact same code in every realm auditor. We can also introduce other abstract base classes for the client, IdP, and scope auditors.

Reply:
You can always override the `should_consider_realm` method in an actual auditor implementation (or extend it etc.).
Since we have the pattern everywhere that we loop over all entities, check whether they are ignored and then perform the actual check, we can introduce it everywhere imho.

I've created #42.